src/jmc/compile/command/builtin_function/jmc_command.py

```diff
@@ -601,3 +603,284 @@ def call(self) -> str:
+            'minecraft/advancement/nether/summon_wither',
+            'minecraft/advancement/nether/all_potions',
+            'minecraft/advancement/nether/create_beacon',
+            'minecraft/advancement/nether/all_effects',
+            'minecraft/advancement/nether/create_full_beacon',
+            'minecraft/advancement/nether/find_bastion',
+            'minecraft/advancement/nether/obtain_ancient_debris',
+            'minecraft/advancement/nether/obtain_crying_obsidian',
+            'minecraft/advancement/nether/distract_piglin',
+            'minecraft/advancement/nether/ride_strider',
+            'minecraft/advancement/nether/loot_bastion',
+            'minecraft/advancement/nether/use_lodestone',
+            'minecraft/advancement/nether/netherite_armor',
+            'minecraft/advancement/nether/charge_respawn_anchor',
+            'minecraft/advancement/nether/explore_nether',
+            'minecraft/advancement/story/root',
+            'minecraft/advancement/story/mine_stone',
+            'minecraft/advancement/story/upgrade_tools',
+            'minecraft/advancement/story/smelt_iron',
+            'minecraft/advancement/story/obtain_armor',
+            'minecraft/advancement/story/lava_bucket',
+            'minecraft/advancement/story/iron_tools',
+            'minecraft/advancement/story/deflect_arrow',
+            'minecraft/advancement/story/form_obsidian',
+            'minecraft/advancement/story/mine_diamond',
+            'minecraft/advancement/story/enter_the_nether',
+            'minecraft/advancement/story/shiny_gear',
+            'minecraft/advancement/story/enchant_item',
+            'minecraft/advancement/story/cure_zombie_villager',
+            'minecraft/advancement/story/follow_ender_eye',
+            'minecraft/advancement/story/enter_the_end'
+        ]
+        files = list(self.datapack.defined_file_pos.keys())
+
+        if namespace == "minecraft":
+            minecraft_files: list[str] = ['/'.join(i.split("/")[2:]) for i in files + minecraft_advancements if i.split("/")[0] == "minecraft"]
+            if advancement not in minecraft_files:
+                raise JMCFileNotFoundError(f"'{advancement}' advancement in '{namespace}' is not defined or missing")
+        else:
+            namespace_files: list[str] = ['/'.join(i.split("/")[1:]) for i in files] 
+            if advancement not in namespace_files:
+                raise JMCFileNotFoundError(f"'{advancement}' advancement in '{namespace}' is not defined or missing")
+
+        if (not(selection in ["everything","from","only","through","until"])):
+            raise JMCValueError(
+                f"'{selection}' is not an valid argument",
+                self.raw_args["selection"].token,
+                self.tokenizer,
+                suggestion="valid arguments are: everything,from,only,through,until"
+            )
+            
+        if (selection == "everything" and advancement != ""):
```

Comment:
`if` in python doesn't require parenthesis except when there is newline
You can use `if selection == "everything" and not advancement:` or `if selection == "everything" and advancement != "":` 

This applies to every `if` in the code